src/agentscope/rag/knowledge_bank.py

```diff
@@ -0,0 +1,186 @@
+# -*- coding: utf-8 -*-
+"""
+Knowledge bank for making Knowledge objects easier to use
+"""
+import copy
+import json
+from typing import Optional, Union
+from loguru import logger
+from agentscope.models import load_model_by_config_name
+from agentscope.agents import AgentBase
+from .llama_index_knowledge import LlamaIndexKnowledge
+
+
+DEFAULT_INDEX_CONFIG = {
+    "knowledge_id": "",
+    "data_processing": [],
+}
+DEFAULT_LOADER_CONFIG = {
+    "load_data": {
+        "loader": {
+            "create_object": True,
+            "module": "llama_index.core",
+            "class": "SimpleDirectoryReader",
+            "init_args": {},
+        },
+    },
+}
+DEFAULT_INIT_CONFIG = {
+    "input_dir": "",
+    "recursive": True,
+    "required_exts": [],
+}
+
+
+class KnowledgeBank:
+    """
+    KnowledgeBank enables
+    1) provide an easy and fast way to initialize the Knowledge object;
+    2) make Knowledge object reusable and sharable for multiple agents.
+    """
+
+    def __init__(
+        self,
+        configs: Union[dict, str],
+    ) -> None:
+        """initialize the knowledge bank"""
```

Comment:
Should we add some docs to show what fields are required in this config here? (e.g. ['knowledge_id"], ["emb_model_config_name"], etc.)

Reply:
We have added remarks regarding such a format in the tutorial.